OpenSim/Analyses/OutputReporter.cpp

```diff
@@ -0,0 +1,129 @@
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2017 Stanford University and the Authors                *
+ * Author(s):                                                                 *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include "OutputReporter.h"
+#include <OpenSim/Simulation/Model/Model.h>
+
+using namespace OpenSim;
+using namespace std;
+
+//=============================================================================
+// ANALYSIS
+//=============================================================================
+int OutputReporter::begin(SimTK::State& s)
+{
+    if (!proceed()) return(0);
+
+    _model = _model->clone();
+
+    _tableReporterDouble = new TableReporter_<double>();
+    _tableReporterVec3 = new TableReporter_<SimTK::Vec3>();
+    _tableReporterSpatialVec = new TableReporter_<SimTK::SpatialVec>();
+
+    _model->addComponent(_tableReporterDouble.get());
+    _model->addComponent(_tableReporterVec3.get());
+    _model->addComponent(_tableReporterSpatialVec.get());
+
+    for (int i = 0; i < getProperty_output_paths().size(); ++i) {
+        auto outpath = ComponentPath(get_output_paths(i));
+        // assume that Outputs are for the Model
+        string compName = _model->getAbsolutePathString();
+        // if a a ComponentPath is specified then use the parent path
+        if (outpath.getNumPathLevels()) {
+            compName = outpath.getParentPathString();
+        }
+        auto outputName = outpath.getComponentName();
+
+        const Component* comp = _model;
+        if (!compName.empty())
+            comp = &_model->getComponent(compName);
```

Comment:
This is a little confusing; it would be helpful to add a comment here. I had to read this a few times to understand that you're handling the case that the output is one of the model's outputs (e.g., kinetic energy).